### ISSUE TITLE
framework 4.6 log4net 只记录error日志

### ISSUE THREAD

**Issue by knowyi** [exceptionless/Exceptionless#853]
Hello, my log4net can upload all log level logs in NETCORE environment. Historical project or framework version 4.6, but only error logs can be uploaded. What's the difference?

![image](https://user-images.githubusercontent.com/7392434/109252834-6ba68700-7829-11eb-8f6b-90f3c1c49f1b.png)

<?xml version="1.0" encoding="utf-8" ?>
<log4net debug="true">
	<appender name="DebugAppender" type="log4net.Appender.RollingFileAppender">
		<filter type="log4net.Filter.LevelRangeFilter">
			<levelMin value="DEBUG" />
			<levelMax value="DEBUG" />
		</filter>
		<file value="D:\logs\testProduct\\debug.txt" />
		<appendToFile value="true" />
		<rollingStyle value="Date" />
		<datePattern value="yyyyMMdd" />
		<layout type="log4net.Layout.PatternLayout">
			<conversionPattern value="%date [%thread] %-5level %logger - %message%newline" />
		</layout>
	</appender>
	<appender name="LogAppender" type="log4net.Appender.RollingFileAppender">
		<filter type="log4net.Filter.LevelRangeFilter">
			<levelMin value="INFO" />
			<levelMax value="INFO" />
		</filter>
		<file value="D:\logs\testProduct\\log.txt" />
		<appendToFile value="true" />
		<rollingStyle value="Date" />
		<datePattern value="yyyyMMdd" />
		<layout type="log4net.Layout.PatternLayout">
			<conversionPattern value="%date [%thread] %-5level %logger - %message%newline" />
		</layout>
	</appender>
	<appender name="ExceptionAppender" type="log4net.Appender.RollingFileAppender">
		<filter type="log4net.Filter.LevelRangeFilter">
			<levelMin value="WARN" />
			<levelMax value="FATAL" />
		</filter>
		<file value="D:\logs\testProduct\\Error.txt" />
		<appendToFile value="true" />
		<rollingStyle value="Date" />
		<datePattern value="yyyyMMdd" />
		<layout type="log4net.Layout.PatternLayout">
			<conversionPattern value="%date %message%newline" />
		</layout>
	</appender>
	<appender name="exceptionless" type="Exceptionless.Log4net.ExceptionlessAppender,Exceptionless.Log4net" />
	<root>
		<level value="ALL" />
		<appender-ref ref="DebugAppender" />
		<appender-ref ref="LogAppender" />
		<appender-ref ref="ExceptionAppender" />
		<appender-ref ref="exceptionless"/>
	</root>
</log4net>

**Comment by knowyi** [comment]
<?xml version="1.0" encoding="utf-8" ?>
<log4net debug="true">
	<appender name="DebugAppender" type="log4net.Appender.RollingFileAppender">
		<filter type="log4net.Filter.LevelRangeFilter">
			<levelMin value="DEBUG" />
			<levelMax value="DEBUG" />
		</filter>
		<file value="D:\logs\testProduct\\debug.txt" />
		<appendToFile value="true" />
		<rollingStyle value="Date" />
		<datePattern value="yyyyMMdd" />
		<layout type="log4net.Layout.PatternLayout">
			<conversionPattern value="%date [%thread] %-5level %logger - %message%newline" />
		</layout>
	</appender>
	<appender name="LogAppender" type="log4net.Appender.RollingFileAppender">
		<filter type="log4net.Filter.LevelRangeFilter">
			<levelMin value="INFO" />
			<levelMax value="INFO" />
		</filter>
		<file value="D:\logs\testProduct\\log.txt" />
		<appendToFile value="true" />
		<rollingStyle value="Date" />
		<datePattern value="yyyyMMdd" />
		<layout type="log4net.Layout.PatternLayout">
			<conversionPattern value="%date [%thread] %-5level %logger - %message%newline" />
		</layout>
	</appender>
	<appender name="ExceptionAppender" type="log4net.Appender.RollingFileAppender">
		<filter type="log4net.Filter.LevelRangeFilter">
			<levelMin value="WARN" />
			<levelMax value="FATAL" />
		</filter>
		<file value="D:\logs\testProduct\\Error.txt" />
		<appendToFile value="true" />
		<rollingStyle value="Date" />
		<datePattern value="yyyyMMdd" />
		<layout type="log4net.Layout.PatternLayout">
			<conversionPattern value="%date %message%newline" />
		</layout>
	</appender>
	<appender name="exceptionless" type="Exceptionless.Log4net.ExceptionlessAppender,Exceptionless.Log4net" />
	<root>
		<level value="ALL" />
		<appender-ref ref="DebugAppender" />
		<appender-ref ref="LogAppender" />
		<appender-ref ref="ExceptionAppender" />
		<appender-ref ref="exceptionless"/>
	</root>
</log4net>

**Comment by ejsmith** [comment]
The client sets the default minimum log level to error. You will need to set the min log level that you want either through the UI or in code.

**Comment by knowyi** [comment]
> The client sets the default minimum log level to error. You will need to set the min log level that you want either through the UI or in code.

Is the minimum log level set in log4net configuration file useless?

**Comment by ejsmith** [comment]
I don't think you can set the min level on the appender itself, can you? I don't remember how it's configured. Can you show me a sample?

**Comment by ejsmith** [comment]
Found a sample here: https://stackoverflow.com/q/8926409/55890

So the problem is that generally Exceptionless is used for error reporting and not as a general log collector. With the log level defaulting to all, people were eating up their plan limits unintentionally. So we changed the default min level to error. Since that minimum level is in the Exceptionless client it will override the level set in the log4net appender. There is no way for us to know if you intentionally set a min level in the appender config or not. So we are taking the less chatty default since generally people just want their error log messages sent to exceptionless.

**Comment by knowyi** [comment]
> Found a sample here: https://stackoverflow.com/q/8926409/55890
> 
> So the problem is that generally Exceptionless is used for error reporting and not as a general log collector. With the log level defaulting to all, people were eating up their plan limits unintentionally. So we changed the default min level to error. Since that minimum level is in the Exceptionless client it will override the level set in the log4net appender. There is no way for us to know if you intentionally set a min level in the appender config or not. So we are taking the less chatty default since generally people just want their error log messages sent to exceptionless.

I wonder why NETCORE's log4net will upload all log levels

**Comment by knowyi** [comment]
> I don't think you can set the min level on the appender itself, can you? I don't remember how it's configured. Can you show me a sample?

I tested the Framework 4.6. The minimum level of log upload is warn. I want to change it to info. I don't know where to set it
